### DISCUSSION TITLE
Updating note in frontend after changing properties in script

### DISCUSSION THREAD

**Discussion by xyrob** [TriliumNext/Trilium#6052] [Ideas]
I'm having trouble figuring out what the right way to change properties (like note.title) from a script.  I can change the property on the backend, but that doesn't notify anything that the note has actually changed.  To change the note title, I'm trying this:

```
   async setNoteTitle(noteId, title) {
        await api.runOnBackend((noteId, title) => {
            const note = api.getNote(noteId);
            note.title = title;
        }, [noteId, title]);
        
        await api.waitUntilSynced();

        await api.reloadNotes([noteId]);
    }
```

This was just a guess based on some other scripts and doesn't work correctly.  The title doesn't update on the frontend unless I reload (Ctrl+R) the electron app, but even then I think there are some synchronization conflicts since I've had the changes get undone sometimes.

What's the right way to modify a note and tell everything that the note has changed?

**Comment by maphew** [comment]
I think you need to use `patch` from the ETAPI. At least that's what's needed when using _trilium-py_. 
- https://github.com/Nriver/trilium-py?tab=readme-ov-file#-update-note

I don't know what the javascript equivalent is. Wild ass guess:

```js
...
// get the note object we want to change
const note = api.getNote(noteId);
// assign new title to the note object and update
api.patchNote(note.noteId {title: "new title"});
...
```
